src/physics/em/detail/BetheBlochInteractor.i.hh

```diff
@@ -0,0 +1,203 @@
+        epsilon = std::exp(ln_epsilon);
+        func2 = epsilon * this->differential_cross_section(epsilon, element);
+    } while(func2 < func1 * p(rng));
+
+    real_type gamma_energy = epsilon;
+
+    // Sample secondary direction.
+    UniformRealDistribution<real_type> phi(0, 2 * constants::pi);
+
+    real_type cost = this->sample_cos_theta(gamma_energy, rng);
+    Real3 gamma_dir = rotate(from_spherical(cost, phi(rng)), inc_direction_);
+
+    real_type tot_momentum = std::sqrt(gamma_energy *
+                                    (gamma_energy + 2.0 * inc_mass_.value()));
+    Real3 inc_direction;
+    for (int i = 0; i < 3; i++)
+    {
+        inc_direction[i] = tot_momentum * inc_direction_[i]
+                                        - gamma_energy * gamma_dir[i];
+    }
+    normalize_direction(&inc_direction);
+
+    // Construct interaction for change to primary (incident) particle
+    Interaction result;
+    result.action      = Action::scattered;
+    result.energy      = units::MevEnergy{kinetic_energy - gamma_energy};
+    result.direction   = inc_direction;
+    result.secondaries = {secondaries, 1};
+
+    // Save outgoing secondary data
+    secondaries[0].particle_id = shared_.gamma_id;
+    secondaries[0].energy      = units::MevEnergy{gamma_energy};
+    secondaries[0].direction   = gamma_dir;
+
+    return result;
+}
+
+template<class Engine>
+CELER_FUNCTION real_type
+BetheBlochInteractor::sample_cos_theta(real_type gamma_energy,
+                                       Engine& rng)
+{
+    real_type mass = inc_mass_.value();
+    real_type gam  = 1.0 + inc_energy_.value() / mass;
+    real_type rmax = gam * constants::pi * 0.5
+                         * min(1.0, gam * mass / gamma_energy - 1);
+
+    UniformRealDistribution<real_type> xi;
+    real_type x = xi(rng) * ipow<2>(rmax) / (1 + ipow<2>(rmax));
+
+    return std::cos(std::sqrt(x / (1.0 - x)) / gam);
+}
+
+CELER_FUNCTION real_type
+BetheBlochInteractor::differential_cross_section(real_type gamma_energy,
+                                                 ElementView element)
+{
+
+    real_type dxsection = 0.0;
+    real_type ep_max = inc_energy_.value();
+
+    if (gamma_energy >= ep_max)
+    {
+        return dxsection;
+    }
+
+    int Z = element.atomic_number();
+    real_type A = element.atomic_mass().value();
+    real_type mass = inc_mass_.value();
+    real_type sqrte = 1.648721271;
+    real_type E = mass + ep_max;
+    real_type v = gamma_energy / E;
+    real_type delta = ipow<2>(mass) * v / (2.0 * E - gamma_energy);
+
+    real_type dn, dnstar, b, b1;
+    dn = 1.54 * std::pow(A,0.27);
+
+    if (Z == 1)
+    {
+        dnstar = dn;
+        b = 202.4;
+        b1 = 446.0;
+    }
+    else
+    {
+        dnstar = dn / std::pow(dn,1.0 / Z);
+        b = 183.0;
+        b1 = 1429.0;
+    }
+
+    real_type phi_n;
+    real_type Z13 = 1.0 / element.cbrt_z();
+    real_type electron_m = constants::electron_mass;
+
```

Comment:
This should be the electron mass in MeV/c^2, right? You'll probably need to add an `electron_mass` field to `BetheBlochInteractorPointers`, and then assign its value in the model class's constructor, like: `interface_.electron_mass = particles.get(particles.find(pdg::electron())).mass()` (see for example the Seltzer-Berger pointers and model). Then you can use `shared_.electron_mass` here.